list-doi-requests/src/main/java/no/unit/nva/doirequest/list/ListDoiRequestsHandler.java

```diff
@@ -85,22 +111,67 @@ private boolean userHasRequestedRole(String role, String userRolesCsv) {
 
     private Publication[] fetchDoiRequestsForUser(UserInstance userInstance) {
         List<DoiRequest> doiRequests = doiRequestService.listDoiRequestsForUser(userInstance);
-        return convertInternalObjectsToDtos(doiRequests);
+        var publicationDtos = convertInternalObjectsToDtos(doiRequests);
+        List<Publication> enrichedWithMessages = enrichPublicationDtosWithDoiRequestMessages(publicationDtos);
+        return convertListToArray(enrichedWithMessages);
     }
 
     private Publication[] fetchDoiRequestsForCurator(UserInstance userInstance) {
         List<DoiRequest> doiRequests = doiRequestService.listDoiRequestsForPublishedPublications(userInstance);
+        List<Publication> publicationDtos = convertInternalObjectsToDtos(doiRequests);
```

Comment:
Here you use `List<Publication> publicationDtos` that does the same thing as the method above, but the method above uses `var publicationDtos.` And can you reuse the logic as most of the content in the `fetchDoiRequests..` methods share the same logic except for `List<DoiRequest> doirequests` ?